src/components/UserInfo.vue

```diff
@@ -156,82 +164,84 @@ export default {
+        this.user.id,
+        this.selectedClientId
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     getUserAvailableClientRoles: function() {
-      UsersRepository.getUserAvailableClientRoles(
+      return UsersRepository.getUserAvailableClientRoles(
         this.user.id,
         this.selectedClientId
-      )
-        .then(response => {
-          this.availableClientRoles.push(...response.data);
-        })
-        .catch(e => {
-          console.log(e);
-        });
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     getUserEffectiveClientRoles: function() {
-      UsersRepository.getUserEffectiveClientRoles(
+      return UsersRepository.getUserEffectiveClientRoles(
         this.user.id,
         this.selectedClientId
-      )
-        .then(response => {
-          this.effectiveClientRoles.push(...response.data);
-          this.selectedRoles.push(...response.data);
-        })
-        .catch(e => {
-          console.log(e);
-        });
+      ).catch(e => {
+        console.log(e);
+      });
     },
 
     updateUserClientRoles: function() {
-      //If in effective but not selected DELETE
-      var rolesToDelete = this.effectiveClientRoles.filter(
+      //If in roles but not selected DELETE
+      var rolesToDelete = this.clientRoles.filter(
         value => !this.selectedRoles.includes(value)
       );
-
-      //If in available and selected ADD
-      var rolesToAdd = this.availableClientRoles.filter(value =>
+      //If in roles and selected ADD
+      var rolesToAdd = this.clientRoles.filter(value =>
         this.selectedRoles.includes(value)
       );
 
       this.successMessage = "";
       this.errorMessage = "";
 
-      //TODO refactor this
-      UsersRepository.deleteUserClientRoles(
-        this.user.id,
-        this.selectedClientId,
-        rolesToDelete
-      )
-        .then(response => {
-          console.log(response);
-
-          return UsersRepository.addUserClientRoles(
-            this.user.id,
-            this.selectedClientId,
-            rolesToAdd
-          );
-        })
-        .then(response => {
-          console.log(response);
-
-          this.successMessage =
+      Promise.all([
+        UsersRepository.deleteUserClientRoles(
+          this.user.id,
+          this.selectedClientId,
+          rolesToDelete
+        ),
+        UsersRepository.addUserClientRoles(
+          this.user.id,
+          this.selectedClientId,
+          rolesToAdd
+        )
+      ]).then(() => {
+        this.getUserClientRoles();
+        this.successState = true;
+        this.errorState = false;
+        this.successMessage =
             this.successMessage + "Roles Updated Successfully ";
-
-          this.getUserClientRoles();
-        })
-        .catch(error => {
+        window.scrollTo(0, 0);    
+      }).catch(error => {
           this.errorMessage = this.errorMessage + "Error Updating Roles";
           console.log(error);
```

Comment:
I think we should always display a simple error message to the user. I'm familiar with proper client-side error handling and logging, but from a user experience it's nice to see an alert when something goes wrong.

For testing and implementation, I suggest you go to one of the Repositories and add an error:

```javascript
deleteUserClientRoles(userId, clientId, content) 
    let a = true;
    if (a) throw Error('bad stuff')
    //....
}

Then you can test the behaviour of error handling.